libraries/core_libs/consensus/src/pbft/pbft_chain.cpp

```diff
@@ -107,6 +114,9 @@ void PbftChain::updatePbftChain(blk_hash_t const& pbft_block_hash, bool null_anc
   size_++;
   if (!null_anchor) {
     non_empty_size_++;
+    if ((dag_expiry_period_ == 0 && non_empty_size_ > kDagExpiryPeriodLimit) || dag_expiry_period_ > 0) {
```

Comment:
This should not right. Like non_empty_size_ is 2000 and faucet down at the moment. There may still some DAG blocks in the last DAG level as sibling for non finalized. If no new DAG block generate, these DAG blocks will be treat as expired.